lib/hexpm/web/session.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Hexpm.Web.Session do
+  alias Hexpm.Accounts.Session
+  alias Hexpm.Repo
+
+  @behaviour Plug.Session.Store
+
+  def init(_opts) do
+    :ok
+  end
+
+  def get(_conn, cookie, _opts) do
+    {id, token} = parse_cookie(cookie)
+    session = Repo.get(Session, id)
+    if session && Plug.Crypto.secure_compare(token, session.token) do
+      {id, session.data}
+    else
+      {nil, %{}}
+    end
+  end
+
+  def put(_conn, nil, data, _opts) do
+    session = Repo.insert!(Session.build(data))
+    build_cookie(session)
+  end
+
+  def put(_conn, id, data, _opts) do
+    session =
+      if session = Repo.get(Session, id) do
+        Repo.update!(Session.update(session, data))
+      else
+        Repo.insert!(Session.build(data))
+      end
+    build_cookie(session)
+  end
+
+  def delete(_conn, id, _opts) do
+    if session = Repo.get(Session, id) do
```

Comment:
Use `Repo.delete_all`, this way you don't need to do a query to check for the session data.